conans/client/graph/graph_binaries.py

```diff
@@ -60,33 +63,37 @@ def _evaluate_clean_pkg_folder_dirty(self, node, package_layout, pref):
                 package_layout.package_remove()
                 return
 
-    def _evaluate_cache_pkg(self, node, package_layout, pref, remote, remotes, update):
+    def _evaluate_cache_pkg(self, node, pref, remote, remotes, update):
         if update:
             output = node.conanfile.output
             if remote:
                 try:
-                    tmp = self._remote_manager.get_package_manifest(pref, remote)
-                    upstream_manifest, pref = tmp
+                    # if there's a later package revision in the remote we will take that one
+                    pkg_id = PackageReference(pref.ref, pref.id)
+                    remote_prevs = self._remote_manager.get_package_revisions(pkg_id, remote)
```

Comment:
We need to benchmark this a bit, this could be relatively slow and scale poorly on the server side.